outlines/routing.md

```diff
@@ -0,0 +1,49 @@
+# Routing
+
+* What is routing, and how it's different in a client-rendered app
+    * Uses for the URL
+        * Bookmarking
+        * Sharing
+        * Back/forward
+    * It's a useful serialization of some share-able, bookmarkable client-side state
+        * Not everything needs to be a route - consider that you can just store JavaScript variables, or use local storage for temporary information
+    * Routing is not related to data loading or authorization in a client-side app. For those, go see the security and data loading chapters
+* Flow Router intro
+* Creating routes
+    * How to define a basic route
+    * How to accept route parameters and URL pattern matching
+    * When to use query parameters vs. path parameters
+* Getting information about the current route
+    * Be careful about where you do this, the best way is to have the page component and layouts responsible for this
+        * Otherwise, complications can arise when doing transitions between pages, since the parameters are changing
+    * Getting the currently active route
+    * Getting the parameters
+    * Highlighting the currently active route
+* Using the router to display templates/pages
```

Comment:
Mention how to have pages that aren't accessible to certain users, and how to ask people to log in to those pages.